src/main/java/org/jenkinsci/plugins/codesonar/services/HubVersionService.java

```diff
@@ -0,0 +1,181 @@
+package org.jenkinsci.plugins.codesonar.services;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.Consts;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.ParseException;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.util.EntityUtils;
+import org.jenkinsci.plugins.codesonar.CodeSonarPublisher;
+import org.jenkinsci.plugins.codesonar.models.CodeSonarHubInfo;
+import org.jenkinsci.plugins.codesonar.models.VersionCompatibilityInfo;
+
+import com.google.common.base.Throwables;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+import hudson.AbortException;
+
+public class HubVersionService {
```

Comment:
I think you should rename this class from `HubVersionService` to `HubInfoService` to emphasize that its job is to get information about the hub.  The fact that it gets a version string is only a detail, not the general purpose of this class.